tests/test_modeling.py

```diff
@@ -175,4 +175,78 @@ def test_raise_error_not_causallm(self):
         # This should raise a ValueError
         with self.assertRaises(ValueError):
             pretrained_model = AutoModelForCausalLM.from_pretrained(model_id)
-            _ = AutoModelForCausalLMWithValueHead.from_pretrained(pretrained_model.transformer)
\ No newline at end of file
+            _ = AutoModelForCausalLMWithValueHead.from_pretrained(pretrained_model.transformer)
+
+
+class ReferenceModelTest(unittest.TestCase):
+
+    def setUp(self):
+        self.model = AutoModelForCausalLMWithValueHead.from_pretrained("hf-internal-testing/tiny-random-GPT2Model")
```

Comment:
Small nit: 
```suggestion
        self.model = AutoModelForCausalLMWithValueHead.from_pretrained("trl-internal-testing/tiny-random-GPT2Model")
```
Let's use our org instead :D